### PR TITLE
Fill in `Overview.md`

### PR DESCRIPTION
**Pull request by fitzgen**
This fills in `Overview.md` with content from the design issue, CG presentation, and subsequent discussions.

It tentatively specifies that custom page sizes must be a power of two between 1 and 64KiB. It is worth noting, however, that there is still somewhat active discussion surrounding what the exact values allowed should be. Powers of two seems like the option with the most support currently. It is also a superset of the other options so, should we decide to restrict custom page sizes to only the values 1 and 64KiB in the future, for example, it should be an easy migration. See https://github.com/WebAssembly/custom-page-sizes/issues/2 for more details.

There is a section that summarizes the discussion in https://github.com/WebAssembly/custom-page-sizes/issues/3. I think we've resolved the toolchain-related open questions we've thought of thus far, so I will go ahead and say that this commit

Fixes https://github.com/WebAssembly/custom-page-sizes/issues/3

Finally, there is a new section on binary encoding. I have not attempted to sketch this out yet. I will open a dedicated issue for discussing this subject.

### CODE REVIEW

**Comment by fitzgen** [comment]
[Here is the rendered `Overview.md`](https://github.com/WebAssembly/custom-page-sizes/blob/fill-in-overview/proposals/custom-page-sizes/Overview.md)

I'll leave this PR open for a few days for folks to comment on it and provide any feedback they have before merging.

File: proposals/custom-page-sizes/Overview.md [hunk 1]
```diff
@@ -1,3 +1,294 @@
...
+  ;; size is 1.
+  (func (export "get_defined_memory_size_in_bytes") (result i32)
+    memory.size $defined
+  )
+)
+```
+
+### Binary Encoding
+
 TODO
```
Comment (alexcrichton) [inline proposals/custom-page-sizes/Overview.md:131]: Not relevant to this PR, but for the future PR to fill this in I found [this table in the memory64 proposal](https://github.com/WebAssembly/memory64/blob/main/proposals/memory64/Overview.md#binary-format) perhaps a good starting point for determining a binary encoding. (aka pick the next bit in the prefix byte and have it represent a "trailing log2 page size i32" and that may be all that's necessary)

**Comment by fitzgen** [comment]
> Finally, there is a new section on binary encoding. I have not attempted to sketch this out yet. I will open a dedicated issue for discussing this subject.

Opened https://github.com/WebAssembly/custom-page-sizes/issues/5 for this discussion

File: proposals/custom-page-sizes/Overview.md (hunk 1 above)
Comment (fitzgen) [reply proposals/custom-page-sizes/Overview.md:131]: Thanks for the pointer and the reminder to file an issue. Just did so: https://github.com/WebAssembly/custom-page-sizes/issues/5

**Comment by fitzgen** [comment]
> I'll leave this PR open for a few days for folks to comment on it and provide any feedback they have before merging.

Looks like there is not any substantive feedback on this overview document, so I'll go ahead and merge it now.

**Pull request merged**